### PR TITLE
Forgotten Flatpacks

### PR DESCRIPTION
**Pull request by Kurfursten**
Remember back when you had to do a bunch of funny steps to unfold flatpacks? Kurfurst remembers.

Traders now peddle 2 such ancient flatpacks, which contain ancient, now-banned technologies.

**Condiment Dispenser** - Dispenses salt, pepper, ketchup, mustard, relish, hotsauce, coldsauce, butter, soysauce, and sprinkles... directly onto food! (or into a condiment bottle) - it has a max energy of 30, though, so don't go crazy. It even has a custom template and sprite to reflect this.
**Electrolytic ChemMaster** - Annoyed that you have to slum it with a grenade casing to do ghetto chem tricks as a chemist? This baby can do it right in the chemmaster, pulling power from the APC instead of a cell, and it has a noreact reservoir so no juggling cans. (You can only electrolyze something from the inserted container into the reservoir)

Tested. They're both fun to play with.

:cl:
* rscadd: Traders are now selling a condiment dispenser and an electrolytic chemmaster. Some assembly required!

### CODE REVIEW

**Comment by dunsel-hordon** [comment]
so can these machines be scanned and then made?

**Comment by Kurfursten** [comment]
I didn't give them the mechscanfail flag, so you can scan them. Do you think they should be forbidden from duplication?

**Comment by Irradiation** [comment]
They're fairly niche machinery that'll seldom get used, so I don't see harm in making them scannable.

**Comment by DrSnips** [comment]
Could you elaborate on the "some assembly required" part? Asking for a friend.

**Comment by Kurfursten** [comment]
It's just like the old mechanic flatpacks from when they first came out. You get a randomized list of things you have to do to actually convert the flatpack into a machine, which are all tool operations (e.g.: screwdriver, welder, wrench, wirecutter). Sometimes the list can have misprinted instructions, too.

File: code/modules/research/mechanic/flatpack.dm [hunk 1]
```diff
@@ -57,36 +64,35 @@
...
+						instructions.name = "misprinted " + instructions.name
+					instructions.update_icon()
+					assembling = ASSEMBLING
+				else
+					machine.forceMove(src.loc)
+					machine = null
+					qdel(src)
 			else
-				to_chat(user, "<span class='notice'>[bicon(src)]It seems this [src] was empty...</span>")
+				to_chat(user, "<span class='notice'>[bicon(src)] It seems this [src] was empty...</span>")
```
Comment (DamianX) [inline code/modules/research/mechanic/flatpack.dm:94]:
```suggestion
				to_chat(user, "<span class='notice'>[bicon(src)] It seems this [name] was empty...</span>")
```

**Comment by FudgePucker** [comment]
Ah yes old flat packs. You should add back in the old exploit where you can just spam the tool for the next step to skip the rest of the steps for nostalgia reasons :^)

**Pull request merged**